accounts/abi/bind/base.go

```diff
@@ -232,7 +289,7 @@ func (c *BoundContract) transact(opts *TransactOpts, contract *common.Address, i
 		rawTx = types.NewTransaction(nonce, c.address, value, gasLimit, gasPrice, input)
 	}
 	if opts.Signer == nil {
-		return nil, errors.New("no signer to authorize the transaction with")
+		return rawTx, errors.New("no signer to authorize the transaction with")
```

Comment:
Returning a valid value while also returning an error violates errors handling conventions. Why is this change needed?

Reply:
Ah forgot to unstage this change, nvm.